src/vs/workbench/contrib/chat/common/promptSyntax/contentProviders/filePromptContentsProvider.ts

```diff
@@ -64,14 +64,33 @@ export class FilePromptContentProvider extends PromptContentsProviderBase<FileCh
 		// get the binary stream of the file contents
 		let fileStream;
 		try {
+			// ensure that the referenced URI points to a file before
+			// trying to get a stream for its contents
+			const info = await this.fileService.resolve(this.uri);
+
+			// validate that the cancellation was not yet requested
+			assert(
+				!cancellationToken?.isCancellationRequested,
+				new CancellationError(),
+			);
+
+			assert(
+				info.isFile,
+				new FolderReference(this.uri),
```

Comment:
We throw a folder reference as an error?

Reply:
I see you use that for control flow down below. I would suggest that this check be made explicitly before entering this method. Stuff like this can be kind of fragile over time since consumers need to know about (currently undocumented) behavior of the method and all method it calls in order to use it properly.

Reply:
Yeah, folder references inside prompt files are invalid at this moment.